glean-core/ios/GleanTests/Scheduler/MetricsPingSchedulerTests.swift

```diff
@@ -0,0 +1,334 @@
+            disabled: false
+        )
+        let expectedValue = "must-exist-in-the-first-ping"
+
+        Glean.shared.resetGlean(clearStores: false)
+        expectedStringMetric.set(expectedValue)
+
+        // Destroy Glean, it will retain the recorded metric.
+        Glean.shared.testDestroyGleanHandle()
+        Dispatchers.shared.setTaskQueuing(enabled: true)
+
+        // Create data and attempt to record data before Glean is initialized.  This will
+        // be queued in the dispatcher.
+        let stringMetric = StringMetricType(
+            category: "telemetry",
+            name: "canary_metric",
+            sendInPings: ["metrics"],
+            lifetime: Lifetime.ping,
+            disabled: false
+        )
+        let canaryValue = "must-not-be-in-the-first-ping"
+        stringMetric.set(canaryValue)
+
+        // Set the last time the "metrics" ping was sent to yesterday, which should make
+        // the ping overdue and trigger it at startup.
+        let yesterday = Calendar.current.date(byAdding: Calendar.Component.day, value: -1, to: now)
+        Glean.shared.metricsPingScheduler.updateSentDate(yesterday!)
+
+        let host = URL(string: Configuration.Constants.defaultTelemetryEndpoint)!.host!
+        stub(condition: isHost(host)) { data in
+            let body = (data as NSURLRequest).ohhttpStubs_HTTPBody()
+            let json = try! JSONSerialization.jsonObject(with: body!, options: []) as? [String: Any]
+            XCTAssert(json != nil)
+            let metrics = json?["metrics"] as? [String: Any]
+            let strings = metrics?["string"] as? [String: Any]
+
+            // Ensure there is only the expected metric
+            XCTAssertEqual(1, strings?.count, "Must contain only the expected metric")
+
+            // Check the received metric's value against the expected value
+            let receivedValue = strings?["telemetry.expected_metric"] as? String
+            XCTAssertEqual(expectedValue, receivedValue, "Values must match")
+
+            DispatchQueue.main.async {
+                // let the response get processed before we mark the expectation fulfilled
+                self.expectation?.fulfill()
+            }
+
+            return OHHTTPStubsResponse(
+                jsonObject: [],
+                statusCode: 200,
+                headers: ["Content-Type": "application/json"]
+            )
+        }
+
+        // Set our expectation that will be fulfilled by the stub above
+        expectation = expectation(description: "Metrics Ping Received")
+
+        // Initialize Glean the SECOND time: it will send the expected string metric (stored from
+        // the previous run) but must not send the canary string, which would be sent the next time
+        // the "metrics" ping is collected after this one.
+        // Glean.shared.initialize(uploadEnabled: true)
+        Glean.shared.initialize(uploadEnabled: true)
+        wait(for: [expectation!], timeout: TimeInterval(5.0))
```

Comment:
In other tests we use `waitForExpectations` and provide an error message:

https://github.com/mozilla/glean/blob/master/samples/ios/app/glean-sample-appUITests/DeletionRequestPingTest.swift#L48-L50